src/psyclone/dynamo0p3.py

```diff
@@ -7655,9 +7653,13 @@ def gen_code(self, parent):
             return
 
         # Set halo clean/dirty for all fields that are modified
-        fields = self.unique_modified_args("gh_field")
+        if not self.unique_modified_args("gh_field"):
+            return
 
-        if not fields:
+        if self.ancestor((ACCRegionDirective, OMPRegionDirective)):
```

Comment:
Instead of hard-coding the classes here, would it make sense to add a method to the directive base class `can_include_class` (not sure about the name tbh, but you get the idea). So instead of searching for hard-coded classes, you would have to find a Directive as ancestor, and check if that disables calls inside of the routine.
Just a suggestion, I will still approve the PR without this.

Reply:
Thinking about it, Transformations already have an `excluded_node_types` property and ideally we don't want to duplicate information. Information on forbidden node types could also be used when performing the global validation checks in the backend. Could we generate a Transformations `excluded_node_types` from similar properties from one or more Node types that we know the Transformation inserts?

Reply:
I've opened #1755 to discuss this further.

Reply:
Am happy to postpone this.